From 22c71b6b45e3ce6266e9cc0b4bdc3420d6cb5669 Mon Sep 17 00:00:00 2001
From: Stefan Chulski <stefanc@marvell.com>
Date: Sun, 31 Dec 2017 09:11:08 +0200
Subject: [PATCH 2205/2241] marvell: cp110: dts: enable mvpp2 hw coherency for
 cp110

Remove dma-coherent from SoC level dts files.
Add dma-coherent to chip level cp110 dts file.
this way PPv2 will use HW coherency always.

Change-Id: Ifa25829b247cc1656daaf4202452da46c991c4b5
Signed-off-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/48273
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/49467
Tested-by: Omri Itach <omrii@marvell.com>
---
 arch/arm64/boot/dts/marvell/armada-7020-amc.dts | 1 -
 arch/arm64/boot/dts/marvell/armada-7020.dtsi    | 3 ---
 arch/arm64/boot/dts/marvell/armada-7040.dtsi    | 3 ---
 arch/arm64/boot/dts/marvell/armada-8020.dtsi    | 6 ------
 arch/arm64/boot/dts/marvell/armada-8040.dtsi    | 6 ------
 arch/arm64/boot/dts/marvell/armada-cp110.dtsi   | 1 +
 6 files changed, 1 insertion(+), 19 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-7020-amc.dts b/arch/arm64/boot/dts/marvell/armada-7020-amc.dts
index 65dc43a..2d5a246 100644
--- a/arch/arm64/boot/dts/marvell/armada-7020-amc.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7020-amc.dts
@@ -145,7 +145,6 @@
 			};
 
 			ppv22@000000 {
-				dma-coherent;
 				cp0_eth0: eth0@010000 {
 					status = "okay";
 				};
diff --git a/arch/arm64/boot/dts/marvell/armada-7020.dtsi b/arch/arm64/boot/dts/marvell/armada-7020.dtsi
index 8bc26f7..a6ff9dd 100644
--- a/arch/arm64/boot/dts/marvell/armada-7020.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-7020.dtsi
@@ -156,9 +156,6 @@
 			rtc@284000 {
 				status = "okay";
 			};
-			ppv22@000000 {
-				dma-coherent;
-			};
 			cp0_sata: sata@540000 {
 				#stream-id-cells = <1>;
 			};
diff --git a/arch/arm64/boot/dts/marvell/armada-7040.dtsi b/arch/arm64/boot/dts/marvell/armada-7040.dtsi
index 2f9b258..caef918 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-7040.dtsi
@@ -166,9 +166,6 @@
 			rtc@284000 {
 				status = "okay";
 			};
-			ppv22@000000 {
-				dma-coherent;
-			};
 			cp0_sata: sata@540000 {
 				#stream-id-cells = <1>;
 			};
diff --git a/arch/arm64/boot/dts/marvell/armada-8020.dtsi b/arch/arm64/boot/dts/marvell/armada-8020.dtsi
index 655fbda..9fbcdf4 100644
--- a/arch/arm64/boot/dts/marvell/armada-8020.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-8020.dtsi
@@ -191,9 +191,6 @@
 
 	cp110-0 {
 		config-space {
-			ppv22@000000 {
-				dma-coherent;
-			};
 			pinctrl@440000 {
 				compatible = "marvell,a80x0-cp0-pinctrl";
 
@@ -251,9 +248,6 @@
 
 	cp110-1 {
 		config-space {
-			ppv22@000000 {
-				dma-coherent;
-			};
 			pinctrl@440000 {
 				compatible = "marvell,a80x0-cp1-pinctrl";
 
diff --git a/arch/arm64/boot/dts/marvell/armada-8040.dtsi b/arch/arm64/boot/dts/marvell/armada-8040.dtsi
index f89c1c8..8e89d9d 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-8040.dtsi
@@ -204,9 +204,6 @@
 	cp0 {
 		config-space {
 			/* Temp: disable all Coresight nodes */
-			ppv22@000000 {
-				dma-coherent;
-			};
 			pinctrl@440000 {
 				compatible = "marvell,a80x0-cp0-pinctrl";
 
@@ -264,9 +261,6 @@
 	cp1 {
 		config-space {
 			/* Temp: disable all Coresight nodes */
-			ppv22@000000 {
-				dma-coherent;
-			};
 			pinctrl@440000 {
 				compatible = "marvell,a80x0-cp1-pinctrl";
 
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
index 59b0a4d..154b060 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
@@ -472,6 +472,7 @@
 					<&CP110_LABEL(syscon0) 1 6>, <&CP110_LABEL(syscon0) 1 5>;
 				clock-names = "pp_clk", "gop_core_clk", "gop_clk", "mg_core_clk", "mg_clk";
 				status = "okay";
+				dma-coherent;
 				CP110_LABEL(eth0): eth0@010000 {
 					interrupts = <ICU_GRP_NSR 39 IRQ_TYPE_LEVEL_HIGH>,
 							<ICU_GRP_NSR 43 IRQ_TYPE_LEVEL_HIGH>,
-- 
2.7.4

